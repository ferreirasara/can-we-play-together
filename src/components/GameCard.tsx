import { Card, Box, CardContent, Typography, CardMedia, useTheme, Chip, useMediaQuery } from "@mui/material";

type GameTitleProps = {
  title: string
}
function GameTitle({ title }: GameTitleProps) {
  return (
    <CardContent>
      <Typography variant="h5">{title}</Typography>
    </CardContent>
  )
}

type GameTagsProps = {
  tags: string[]
}
function GameTags({ tags }: GameTagsProps) {
  return (
    <Box
      display="flex"
      flexWrap="wrap"
      gap={1}
      p={1}
    >
      {tags?.map((tag, index) => <Chip label={tag} key={index} />)}
    </Box>
  )
}

type GameImageProps = {
  imageUrl: string
}
function GameImage({ imageUrl }: GameImageProps) {
  return (
    <CardMedia
      component="img"
      sx={{ height: 215 * 0.7, width: 460 * 0.7 }}
      image={imageUrl}
    />
  )
}

export type GameCardProps = {
  title: string
  imageUrl: string
  tags: string[]
  description: string
}
export default function GameCard({ imageUrl, tags, title, description }: GameCardProps) {
  const theme = useTheme()
  const isDesktop = useMediaQuery(theme.breakpoints.up('sm'))

  return (
    <Card sx={{
      display: 'flex',
      flexDirection: isDesktop ? 'row' : 'column-reverse',
      width: '100%',
      justifyContent: 'space-between',
      alignItems: 'center',
      p: 1,
    }}>
      <Box sx={{ display: 'flex', flexDirection: 'column' }}>
        <GameTitle title={title} />
        <GameTags tags={tags} />
      </Box>
      <GameImage imageUrl={imageUrl} />
    </Card>
  )
}