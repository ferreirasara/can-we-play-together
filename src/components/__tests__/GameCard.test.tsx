import renderer from 'react-test-renderer';
import GameCard from '../GameCard';

it('renders correctly', () => {
  const filters = renderer
    .create(
      <GameCard
        appid={0}
        imageUrl={'https://cdn.akamai.steamstatic.com/steam/apps/246620/header.jpg?t=1663763693'}
        tags={['tag1', 'tag2', 'tag3']}
        title={'Game for test'}
        key={'game_test'}
      />
    ).toJSON();
  expect(filters).toMatchSnapshot();
});