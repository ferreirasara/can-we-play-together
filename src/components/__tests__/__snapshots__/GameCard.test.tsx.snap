// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mr8rz7-MuiPaper-root-MuiCard-root"
  onClick={[Function]}
>
  <div
    className="MuiBox-root css-j7qwjs"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Game for test
      </h5>
    </div>
    <div
      className="MuiBox-root css-10x12t9"
    >
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          tag1
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          tag2
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          tag3
        </span>
      </div>
    </div>
  </div>
  <img
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1fn6xi0-MuiCardMedia-root"
    src="https://cdn.akamai.steamstatic.com/steam/apps/246620/header.jpg?t=1663763693"
  />
</div>
`;
